source/core/src/main/com/csse3200/game/components/inventory/PlayerInventoryDisplay.java

```diff
@@ -275,8 +336,16 @@ private void addItem(AbstractItem item) {
     void regenerateInventory() {
         toggleInventory(); // Hacky way to regenerate inventory without duplicating code
         toggleInventory();
+        regenerateHotbar();
```

Comment:
This is unnecessary - please delete. If you would like you can rename it to regenerateDisplay. Toggling the inventory will regenerate the hotbar and inventory display already (in main). See how this is implemented already in main (if you want you can also rename the toggle from main to toggleDisplay) and rework it to match that (that is the only function that needs changing to ensure the hotbar stays on the screen).

Reply:
if I remove this it and go with the main, the hotbar disappears on opening inventory and for drag and drop I want the hotbar to stay on toggling the inventory

Reply:
pls take a look on the function i added a condition